### PR TITLE
Bump to ap-base:3.14

### PR DESCRIPTION
**Pull request by danielhoherd**
## Summary of changes

Alpine 3.14 was merged to ap-base in https://github.com/astronomer/ap-base/pull/21 and all the images that reference it have been updated here

## Images are updated by this PR

- [X] curator
- [X] nginx-es
- [X] postgres-exporter
- [X] registry

## Checklist

Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.

- [X] version.txt is updated in the changed image(s)

### CODE REVIEW

**Review by vishwas-astro — APPROVED** [review]

**Review by andriisoldatenko — APPROVED** [review]

**Comment by danielhoherd** [comment]
We are experiencing a build problem that is likely related to the new alpine version.
- <https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.14.0#faccessat2>
- <https://gitlab.alpinelinux.org/alpine/aports/-/issues/12396>
- <https://app.circleci.com/pipelines/github/astronomer/ap-vendor/456/workflows/e25545a4-b33a-471e-b3d7-58103f0be469/jobs/31152/parallel-runs/0/steps/0-103>

```
 ---> Running in 502edd5a9b3e
make: /bin/sh: Operation not permitted
make: *** [Makefile:101: clean] Error 127
The command '/bin/sh -c CGO_ENABLED=0 make PREFIX=/go clean binaries && file ./bin/registry | grep "statically linked"' returned a non-zero code: 2
```

According to the alpine release notes, it looks like our CircleCI versions of runc and containerd are outdated:
```
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
```

runc >= v1.0.0-rc93 and containerd.io >= 1.4.3-2 are required. I'm not sure if we meet the libseccomp >= 2.4.4.

**Review by andriisoldatenko — APPROVED** [review]

**Comment by danielhoherd** [comment]
Current status is we're still waiting on cimg/python:3.9 to be updated. I have a cron job running `docker pull cimg/python:3.9 | grep -q e6b8443f31fc358c5a1fc1acc259062eefda0450fbb9652ed116e3c914de2813 || prowl "cimg/python:3.9 has been updated https://github.com/astronomer/ap-vendor/pull/110"` a few times daily so I will be alerted whenever this happens.

File: curator/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM quay.io/astronomer/ap-base:3.13.5
+FROM quay.io/astronomer/ap-base:3.14.0
```
Comment (andriisoldatenko) [inline curator/Dockerfile:16]:
```suggestion
FROM quay.io/astronomer/ap-base:3.14.1
```

File: postgres-exporter/Dockerfile [hunk 2]
```diff
@@ -21,7 +21,7 @@ FROM wrouesnel/postgres_exporter:v0.8.0 AS upstream
 # In the next update, I think they are migrating from the scratch image,
 # so it will likely be unnecessary to perform this multi-stage build.
 # https://quay.io/repository/astronomer/ap-base
-FROM quay.io/astronomer/ap-base:3.13.5
+FROM quay.io/astronomer/ap-base:3.14.0
```
Comment (andriisoldatenko) [inline postgres-exporter/Dockerfile:24]:
```suggestion
FROM quay.io/astronomer/ap-base:3.14.1
```

File: postgres-exporter/version.txt [hunk 3]
```diff
@@ -1 +1 @@
-3.13.5
+3.14.0
```
Comment (andriisoldatenko) [inline postgres-exporter/version.txt:1]:
```suggestion
3.14.1
```

File: registry/Dockerfile [hunk 4]
```diff
@@ -36,7 +36,7 @@ RUN git clone https://github.com/distribution/distribution.git $DISTRIBUTION_DIR
 
 RUN CGO_ENABLED=0 make PREFIX=/go clean binaries && file ./bin/registry | grep "statically linked"
 
-FROM quay.io/astronomer/ap-base:3.13.5
+FROM quay.io/astronomer/ap-base:3.14.0
```
Comment (andriisoldatenko) [inline registry/Dockerfile:39]:
```suggestion
FROM quay.io/astronomer/ap-base:3.14.1
```

File: registry/version.txt [hunk 5]
```diff
@@ -1 +1 @@
-3.13.7-1
+3.14.0
```
Comment (andriisoldatenko) [inline registry/version.txt:1]:
```suggestion
3.14.1
```

**Comment by danielhoherd** [comment]
I rebased and built this today and we are still running into errors. The cimg/python:3.9 image has not yet been updated though, so that makes sense. Two of these builds are passing now, but failing CVEs. These CVEs will likely be fixed by alpine 3.14.2

**Comment by danielhoherd** [comment]
I was wrong in my update yesterday. Only registry is failing, and there are no CVE issues (I must have been looking at the wrong pipeline). I've removed registry from this branch so we can merge the rest.

**Comment by danielhoherd** [comment]
FWIW the cimg/python:3.9 image was finally updated with the three components listed above all updated to the needed versions.

```
$ docker images | grep cimg
cimg/python    3.9     20aeb15f82b1   8 days ago    2.14GB
2021-11-16T08:46:39-0800 dho@litten:/home/dho
$ docker run --rm -ti cimg/python:3.9 /bin/bash
circleci@6523ac9eb530:/docker-share$ runc --version
runc version 1.0.2
commit: v1.0.2-0-g52b36a2
spec: 1.0.2-dev
go: go1.16.8
libseccomp: 2.5.1
circleci@6523ac9eb530:/docker-share$ containerd --version
containerd containerd.io 1.4.11 5b46e404f6b9f661a205e28d59c982d3634148f8
```

**Pull request merged**